delphi_esr_msgs/CMakeLists.txt

```diff
@@ -1,45 +1,94 @@
-cmake_minimum_required(VERSION 2.8.3)
 project(delphi_esr_msgs)
 
-add_definitions(-std=c++11)
+set(ROS_VERSION $ENV{ROS_VERSION})
 
-set(CMAKE_CXX_FLAGS "-std=c++0x ${CMAKE_CXX_FLAGS}")
+set(MSG_FILES
+  "EsrEthTx.msg"
+  "EsrStatus1.msg"
+  "EsrStatus2.msg"
+  "EsrStatus3.msg"
+  "EsrStatus4.msg"
+  "EsrStatus5.msg"
+  "EsrStatus6.msg"
+  "EsrStatus7.msg"
+  "EsrStatus8.msg"
+  "EsrStatus9.msg"
+  "EsrTrack.msg"
+  "EsrTrackMotionPower.msg"
+  "EsrValid1.msg"
+  "EsrValid2.msg"
+  "EsrVehicle1.msg"
+  "EsrVehicle2.msg"
+  "EsrVehicle3.msg"
+  "EsrVehicle4.msg"
+  "EsrVehicle5.msg"
+  "TrackMotionPower.msg"
+)
+
+if(${ROS_VERSION} EQUAL 1)
+
+  cmake_minimum_required(VERSION 2.8.3)
 
-find_package(catkin REQUIRED COMPONENTS
+  # Default to C++11
+  if(NOT CMAKE_CXX_STANDARD)
+    set(CMAKE_CXX_STANDARD 11)
+  endif()
+
+  find_package(catkin REQUIRED
+    COMPONENTS
     message_generation
     std_msgs
-)
+  )
 
-## Generate messages in the 'msg' folder
-add_message_files(
-  FILES
-  EsrEthTx.msg
-  # can Tx
-  EsrStatus1.msg
-  EsrStatus2.msg
-  EsrStatus3.msg
-  EsrStatus4.msg
-  EsrStatus5.msg
-  EsrStatus6.msg
-  EsrStatus7.msg
-  EsrStatus8.msg
-  EsrStatus9.msg
-  EsrValid1.msg
-  EsrValid2.msg
-  EsrTrack.msg
-  EsrTrackMotionPower.msg
-  TrackMotionPower.msg
-  # can Rx
-  EsrVehicle1.msg
-  EsrVehicle2.msg
-  EsrVehicle3.msg
-  EsrVehicle4.msg
-  EsrVehicle5.msg
-)
+  add_message_files(FILES
+    ${MSG_FILES}
+    DIRECTORY msg
+  )
 
-generate_messages(DEPENDENCIES std_msgs)
+  generate_messages(DEPENDENCIES std_msgs)
 
-catkin_package(CATKIN_DEPENDS
-  message_runtime
-  std_msgs
-)
+  catkin_package(
+    CATKIN_DEPENDS message_runtime
+  )
+
+elseif(${ROS_VERSION} EQUAL 2)
+
+  cmake_minimum_required(VERSION 3.5)
+
+  if(NOT CMAKE_CXX_STANDARD)
+    set(CMAKE_CXX_STANDARD_REQUIRED ON)
+    set(CMAKE_CXX_STANDARD 14)
+  endif()
+
+  if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+    add_compile_options(-Wall -Wextra -Wpedantic)
+  endif()
+
+  find_package(ament_cmake REQUIRED)
+  find_package(builtin_interfaces REQUIRED)
+  find_package(std_msgs REQUIRED)
+  find_package(rosidl_default_generators REQUIRED)
+
+  # Apend "msg/" to each file name
+  set(TEMP_LIST "")
+  foreach(MSG_FILE ${MSG_FILES})
+    list(APPEND TEMP_LIST "msg/${MSG_FILE}")
```

Comment:
Kinda sucks that this is necessary :neutral_face:, maybe the `rosidl_generate_interfaces` function should also take a `DIRECTORY` option.

Reply:
That would be nice. It wouldn't be too difficult - just editing https://github.com/ros2/rosidl/blob/master/rosidl_cmake/cmake/rosidl_generate_interfaces.cmake to take another parameter. You'd have to work out the details of implementation with the maintainers first, I would imagine.